### DISCUSSION TITLE
referring to Callout blocks in text

### DISCUSSION THREAD

**Discussion by GabriellaS-K** [quarto-dev/quarto-cli#4862] [Q&A]
Hi!
I am writing my thesis in rmarkdown, but would like to do some definitions in boxes. Quarto seems a good way to do this, using the callout blocks. Is there a way to refer to the blocks 'in text'. For example 'for definition, see block 1'? Or even better, a way to save the block, and bring it into my rmarkdown file where the rest of my thesis is? 

Callouts in markdown don't knit to word and pdf nicely :( 

 [link for the callout block](https://quarto.org/docs/authoring/callouts.html)

```
::: {.callout-note icon=false}

## Definintion

Define some important thing

:::

```

**Comment by cscheid** [comment]
>  Is there a way to refer to the blocks 'in text'?

There currently isn't a way to use callouts in crossrefs, unfortunately. I'm going to open an issue to track this as an enhancement for 1.4. We'll be remaking crossrefs entirely. I'm not sure we'll be able to pull it off, but it's an interesting idea.

> Callouts in markdown don't knit to word and pdf nicely :(

I'm not sure what you mean; quarto does offer callout rendering in word and pdf formats. Do you mean in rmarkdown? In that case, yes, I think callouts aren't supported in rmarkdown.

**Reply by GabriellaS-K** [reply]
Yes I meant rmarkdown, my autocorrect keeps changing it! Referencing call outs would be great! I'll keep my fingers crossed :)

**Comment by martmtram** [comment]
Frankly, I have not tested it. But one possible solution is to have i. Python cell inside the callout block, ii. label the block with #| label: and iii. try to reference it in the text. 
I have not tried to reference to it, but I have the callout blocks labeled this way to show up in VS Code outline for navigation.

**Reply by mcanouil** [reply]
Since 1.5, (see the documentation) call-outs can be cross-referenced.

Before posting on a 2 years old thread, it's usually better to search in the documentation / changelog for updates.
